### PR TITLE
Browse items by category card #4

### PR DESCRIPTION
**Pull request by PlanetEfficacy**
@tgisg @kctrlv we re-did our ```:/category_name``` route as well as our has many through relationship between items and categories. We were using ```has_many_belongs_to``` which @case-eee  informed us is not preferred and might be deprecated in certain versions of activerecord.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,58 +1,10 @@
...
-  #   resources :posts, concerns: :toggleable
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  get ':categories', to: 'categories#show', as: :category
```
Comment (tgisg) [inline config/routes.rb]: Any particular reason for this? Why is the regular show path insufficient?

File: db/schema.rb [hunk 2]
```diff
@@ -11,11 +11,22 @@
...
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "categories", force: :cascade do |t|
+    t.string   "name"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "categorizations", id: false, force: :cascade do |t|
```
Comment (tgisg) [inline db/schema.rb]: Can we come up with a better name for the join table? Maybe "collections" or "product_collections" or something? Categorizations seems a bit vague to me.

**Comment by tgisg** [comment]
I've added some comments. The code itself looks good, although I'm not sure why we need to rename the category show path. I find it useful at this stage in my learning to follow a formula for these paths. It's extra mental overhead at this point to remember that we've created a special path. I also mentioned some naming peculiarities, but those are obviously less important.

File: config/routes.rb (hunk 1 above)
Comment (PlanetEfficacy) [inline config/routes.rb]: The spec says we need to specifically get to a category's items through a path called ```/:CATEGORY_NAME```. So we need to see all pencils through path ```/wiriting_implements``` not ```categories/1```. Does that make sense? Do you agree with the implementation?

File: db/schema.rb (hunk 2 above)
Comment (PlanetEfficacy) [inline db/schema.rb]: We struggled with the underscore table join last night / this morning. In particular we weren't getting the pluralization correct across all the various files / classes and we were getting errors. This passes the spec, but I agree it is broadly named. Perhaps we can take a second pass at this if anyone feels confident with rails g migration join tables and can troubleshoot the errors @susiirwin and I ran into last night. Alternatively, we can discuss better names for categorizations table. @tgisg

Comment (tgisg) [inline db/schema.rb]: You and I discussed this, @PlanetEfficacy but I think the syntax would be item_categories or category_items if you wanted to do the plain old nomenclature.

**Comment by tgisg** [comment]
Makes sense now why we use the custom path. :)

**Comment by PlanetEfficacy** [comment]
@tgisg and I changed categorizations table to category_items. All tests passing. Ready to roll.

**Pull request merged**